packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -230,9 +240,18 @@ class ConnectionStateCatchup extends ConnectionStateHandlerPassThrough {
 				this.catchUpMonitor?.dispose();
 				this.catchUpMonitor = undefined;
 				break;
-			case ConnectionState.CatchingUp:
+			// ConnectionState.EstablishingConnection state would only be set when first establishing connection
+			// during Container.load() or container.connect().
+			case ConnectionState.EstablishingConnection:
 				assert(
 					this._connectionState === ConnectionState.Disconnected,
+					"connectivity transition to establishing connection",
+				);
+				break;
+			case ConnectionState.CatchingUp:
+				assert(
+					this._connectionState === ConnectionState.EstablishingConnection ||
```

Comment:
shouldn't we always go through disconnected state first?

Reply:
And we arrive here with ConnectionState.EstablishingConnection, can you please add comment somwhere why / how it happens? For example, is it because of losing connection / hitting error on it (i.e. is it normal path)?